opm/autodiff/SimulatorFullyImplicitBlackoilEbos.hpp

```diff
@@ -148,6 +151,23 @@ class SimulatorFullyImplicitBlackoilEbos
             // This is a restart, populate WellState and ReservoirState state objects from restart file
             output_writer_.initFromRestartFile(props_.phaseUsage(), grid(), state, prev_well_state);
             initHydroCarbonState(state, props_.phaseUsage(), Opm::UgGridHelpers::numCells(grid()), has_disgas_, has_vapoil_);
+            {
```

Comment:
This should preferably be refactored, perhaps into a function 
```initHysteresisState(...)```
I'm open for ideas and suggestions